### PR TITLE
EVG-5297 (feat) Settings service; Persistent perf user settings

### PR DESCRIPTION
**Pull request by unrealsolver**

### CODE REVIEW

File: public/static/app/common/Settings.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+        // Restore default
+        writeSetting(descriptor, value)
+      } else {
+        return undefined
+      }
+    }
+
+    // Type casting
+    return $window[descriptor.type || 'String'](value)
```
Comment (john-m-liu) [inline public/static/app/common/Settings.js]: Why is this needed? Couldn't this just read the value from local storage?

Comment (unrealsolver) [reply public/static/app/common/Settings.js]: @john-m-liu, It does use the local storage. This line is required for type casting. For example, window.Number(valueFromLocalStorage)

Comment (unrealsolver) [reply public/static/app/common/Settings.js]: I use pretty somple type casting approach, I use base JS type classes as both params for a setting type definition and a function/constructor.
Probably, I could remove quotes for type defs and allow use a function.
